src/app/pages/reports-dashboard/components/reports-global-controls/reports-global-controls.component.ts

```diff
@@ -16,82 +20,90 @@ import { waitForPreferences } from 'app/store/preferences/preferences.selectors'
-  @Input() activeTab: ReportTab;
-  @Input() diskMetrics: Option[];
-  @Input() diskDevices: Option[];
-  @Input() selectedDisks: string[];
-  @Input() allTabs: ReportTab[];
-  @Input() activeTabVerified: boolean;
+export class ReportsGlobalControlsComponent implements OnInit {
+  form = this.fb.group({
+    autoRefresh: [false],
+    devices: [[] as string[]],
+    metrics: [[] as string[]],
+  });
 
-  @Output() showConfigForm = new EventEmitter<void>();
-  @Output() navigateToTab = new EventEmitter<ReportTab>();
-  @Output() diskOptionsChanged = new EventEmitter<{ devices: Option[]; metrics: Option[] }>();
-
-  devicesControl: ControlConfig;
-  metricsControl: ControlConfig;
-  autoRefreshEnabled = false;
+  activeTab: ReportTab;
+  selectedDisks: string[];
+  allTabs: ReportTab[] = this.reportsService.getReportTabs();
+  diskDevices$ = this.reportsService.getDiskDevices();
+  diskMetrics$ = this.reportsService.getDiskMetrics();
+  @Output() diskOptionsChanged = new EventEmitter<{ devices: string[]; metrics: string[] }>();
 
   readonly ReportType = ReportType;
 
   constructor(
+    private fb: FormBuilder,
+    private route: ActivatedRoute,
+    private router: Router,
     private store$: Store<AppState>,
-    private translate: TranslateService,
+    private reportsService: ReportsService,
+    private slideIn: IxSlideInService,
   ) {}
 
-  ngOnChanges(): void {
+  ngOnInit(): void {
+    this.activeTab = this.allTabs.find((tab) => tab.value === this.route.routeConfig.path);
     this.setAutoRefreshControl();
-    this.setDiskControls();
+    this.setupDisksTab();
   }
 
-  onShowConfigForm(): void {
-    this.showConfigForm.emit();
+  showConfigForm(): void {
+    this.slideIn.open(ReportsConfigFormComponent);
   }
 
   onNavigateToTab(tab: ReportTab): void {
-    this.navigateToTab.emit(tab);
+    this.router.navigate(['/reportsdashboard', tab.value]);
+  }
+
+  isActiveTab(tab: ReportTab): boolean {
+    return this.activeTab.value === tab.value;
   }
 
-  onDiskOptionsChanged(): void {
-    this.diskOptionsChanged.emit({
-      devices: this.devicesControl.value,
-      metrics: this.metricsControl.value,
+  private setupDisksTab(): void {
+    if (this.activeTab.value !== ReportType.Disk) {
+      return;
+    }
+    if (this.route.snapshot.queryParams?.disks) {
+      this.selectedDisks = this.route.snapshot.queryParams.disks;
+    } else {
+      this.selectedDisks = this.form.value.devices;
+    }
+    this.form.valueChanges.pipe(untilDestroyed(this)).subscribe((values) => {
+      this.diskOptionsChanged.emit({
+        devices: values.devices,
+        metrics: values.metrics,
+      });
     });
+    this.setupDisksToolbar();
   }
 
-  toggleAutoRefresh(): void {
-    this.store$.dispatch(autoRefreshReportsToggled());
+  private setupDisksToolbar(): void {
+    this.diskDevices$.pipe(untilDestroyed(this)).subscribe((diskDevices) => {
+      this.form.patchValue({ devices: diskDevices.map((device) => `${device.value}`) });
```

Comment:
Looks unusual. If you want to convert to string, you can do `String(device.value)`.